phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java

```diff
@@ -633,4 +637,64 @@ public void testJsonQuery() throws Exception {
             compareJson(rs.getString(5), JsonDatatypes, "$.datatypes.mixedArray");
         }
     }
+
+    @Test
+    public void testJsonExpressionIndex() {
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        String tableName = generateUniqueName();
+        String indexName = "IDX_" + generateUniqueName();
+        try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+            conn.setAutoCommit(true);
+            String
+                    ddl =
+                    "create table if not exists " + tableName +
+                            " (pk integer primary key, col integer, jsoncol.jsoncol json)";
+            conn.createStatement().execute(ddl);
+            conn.createStatement().execute(
+                    "UPSERT INTO " + tableName + " (pk, col, jsoncol) VALUES (1,2, '" + JsonDoc1 + "')");
+            conn.createStatement().execute(
+                    "CREATE INDEX " + indexName + " ON " + tableName
+                            + " (JSON_VALUE(JSONCOL,'$.type'), JSON_VALUE(JSONCOL,'$.info.address.town')) include (col)");
```

Comment:
What if we create an index on a json field that is not of type string ? Does the index schema correctly infer that ?

Reply:
Return type of JSON_VALUE is Varchar (String). So all the fields are of type String only and we can use explicit CAST to cast to different types if we need. Do you see some problem here?

Reply:
As I think more about it, it will be fine.